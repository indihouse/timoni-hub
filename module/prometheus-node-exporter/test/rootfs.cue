package main

values: {
	image: {
		digest: "sha256:302028d1ea65a50a32d9ae81fda1a964e45a620a9575e36036b9a55a33395f16"
		tag:    "1.25.2"
	}
	hostRootFs: enabled: true

	podSecurityContext: {
		runAsUser:  65532
		runAsGroup: 65532
		fsGroup:    65532
	}
	securityContext: {
		allowPrivilegeEscalation: false
		readOnlyRootFilesystem:   true
		runAsNonRoot:             true
		capabilities: drop: ["ALL"]
		seccompProfile: type: "RuntimeDefault"
	}
	resources: {
		requests: {
			memory:              "128Mi"
			cpu:                 "100m"
			"ephemeral-storage": "100Mi"
		}
		limits: {
			"ephemeral-storage": "100Mi"
		}
	}
}
