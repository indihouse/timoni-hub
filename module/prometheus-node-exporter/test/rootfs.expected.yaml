---
apiVersion: apps/v1
kind: DaemonSet
metadata:
  labels:
    app.kubernetes.io/name: prometheus-node-exporter
    app.kubernetes.io/version: 0.0.0-devel
  name: prometheus-node-exporter
  namespace: default
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: prometheus-node-exporter
  template:
    metadata:
      labels:
        app.kubernetes.io/name: prometheus-node-exporter
    spec:
      containers:
      - args:
        - --path.sysfs=/host/sys
        - --path.procfs=/host/proc
        - --path.rootfs=/host/root
        image: prom/node-exporter:1.25.2@sha256:302028d1ea65a50a32d9ae81fda1a964e45a620a9575e36036b9a55a33395f16
        imagePullPolicy: IfNotPresent
        name: node-exporter
        ports:
        - containerPort: 9100
          name: metrics
          protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: metrics
        volumeMounts:
        - mountPath: /host/proc
          mountPropagation: HostToContainer
          name: proc
          readOnly: true
        - mountPath: /host/sys
          mountPropagation: HostToContainer
          name: sys
          readOnly: true
        - mountPath: /host/root
          mountPropagation: HostToContainer
          name: root
          readOnly: true
        resources:
          limits:
            ephemeral-storage: 100Mi
          requests:
            cpu: 100m
            ephemeral-storage: 100Mi
            memory: 128Mi
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
          readOnlyRootFilesystem: true
          runAsNonRoot: true
          seccompProfile:
            type: RuntimeDefault
      volumes:
      - hostPath:
          path: /proc
        name: proc
      - hostPath:
          path: /sys
        name: sys
      - hostPath:
          path: /
        name: root
      securityContext:
        fsGroup: 65532
        runAsGroup: 65532
        runAsUser: 65532
---
apiVersion: v1
kind: Service
metadata:
  annotations:
    prometheus.io/port: "9100"
    prometheus.io/scrape: "true"
  labels:
    app.kubernetes.io/name: prometheus-node-exporter
    app.kubernetes.io/version: 0.0.0-devel
  name: prometheus-node-exporter
  namespace: default
spec:
  ports:
  - name: metrics
    port: 9100
    protocol: TCP
    targetPort: metrics
  selector:
    app.kubernetes.io/name: prometheus-node-exporter
  type: ClusterIP
